### PR TITLE
feat: add flow-react module

### PR DESCRIPTION
**Pull request by caalador**
add flow-react into core-internals

Requires vaadin/flow#18710

### CODE REVIEW

**Review by tltv — APPROVED** [review]

File: vaadin-core-internal/pom.xml [hunk 1]
```diff
@@ -43,6 +43,10 @@
             <groupId>com.vaadin</groupId>
             <artifactId>flow-lit-template</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-react</artifactId>
```
Comment (ZheSun88) [inline vaadin-core-internal/pom.xml:48]: should add to the bom for the version management.

Comment (ZheSun88) [reply vaadin-core-internal/pom.xml:48]: pushed the commit

Comment (caalador) [reply vaadin-core-internal/pom.xml:48]: So many places... I sort of expected it to come from the flow-bom

Comment (ZheSun88) [reply vaadin-core-internal/pom.xml:48]: one thing we can start to use in platform is import flow-bom. same as `hilla-bom`

**Pull request merged**